peps/pep-0760.rst

```diff
@@ -0,0 +1,162 @@
+exception-handling syntax. Currently, Python allows catching all exceptions
+with a bare ``except:`` clause, which can lead to overly broad exception
+handling and mask important errors. This PEP suggests requiring explicit
+exception types in all except clauses, promoting more precise and intentional
+error handling.
+
+
+Motivation
+==========
+
+The current syntax allows for catching all exceptions with a bare ``except:`` clause is:
+
+.. code-block:: python
+
+   try:
+       risky_operation()
+   except:
+       handle_any_error()
+
+While this syntax can be convenient for a "catch all" handler, it often leads
+to poor coding practices:
+
+1. It can mask important errors that should be propagated.
+2. It makes debugging more difficult by catching and potentially hiding
+   unexpected exceptions.
+3. It goes against the Python principle of explicit over implicit.
+
+Various linters [1]_ [2]_ [3]_ and style guides (including :pep:`8`) [4]_ [5]_
+[6]_ [7]_ discourage bare ``except`` clauses.
+
+By requiring explicit exception types, we can encourage more thoughtful and
+precise error handling:
+
+.. code-block:: python
+
+   try:
+       risky_operation()
+   except Exception as e:
+       handle_expected_error(e)
+
+
+Rationale
+=========
+
+The decision to disallow bare except clauses is based on the following
+considerations:
+
+1. Requiring specific exception types makes the programmer's intentions clear
+   and encourages thinking about what exceptions might occur.
+
+2. Catching only specific exceptions makes identifying and debugging unexpected
+   errors easier.
+
+3. Preventing overly broad exception handling reduces the risk of silently
+   ignoring critical errors.
+
+4. Many style guides and linters already discourage the use of bare except
+   clauses.
+
+
+Specification
+=============
+
+The syntax for the except clause will be modified to require an exception type.
+The grammar will be updated to remove the possibility of adding an empty
+expression in except clauses.
+
+This change disallows the bare 'except:' syntax. All except clauses must
+specify at least one exception type:
+
+.. code-block:: python
+
+try:
+    ...
+except ValueError:
+    ...
+except (TypeError, RuntimeError):
+    ...
+except Exception:
+    ...  # Still allowed, but catches all exceptions explicitly
+
+The semantics of exception handling remain unchanged, except that it will no
+longer be possible to catch all exceptions without explicitly specifying
+'Exception' or a similarly broad exception type.
+
+
+Backwards Compatibility
+=======================
+
+This change is not backwards compatible. Existing code that uses bare 'except:'
+clauses will need to be modified. To ease the transition:
+
+1. A deprecation warning will be issued for bare except clauses in Python 3.14.
+2. The syntax will be fully disallowed in Python 3.15.
+
+A tool will be provided to automatically update code to replace bare 'except:'
+with 'except Exception:'.
```

Comment:
Why not BaseException?

Reply:
Actually this should say `BaseException`